### ISSUE TITLE
Kernel other than "nearest" has no effect on upscaling?

### ISSUE THREAD

**Issue by einarakerlind** [lovell/sharp#4061]
### What are you trying to achieve?

I noticed that i get identical results for all kernels except nearest when upscaling an image. Is this expected behaviour?
Version: 0.33.3

### When you searched for similar issues, what did you find that might be related?

—

### Please provide a minimal, standalone code sample, without other dependencies, that demonstrates this question

`sharp("path-to-image/example.png").resize({
    width: 600,
    height: 600,
    background: { alpha: 0, r: 0, g: 0, b: 0 },
    kernel: 'lanczos2'
  });`

### Please provide sample image(s) that help explain this question

![example](https://github.com/lovell/sharp/assets/25298812/f7743b30-255f-460d-b077-52f2f65e8afe)
Original image

![example-600-cubic](https://github.com/lovell/sharp/assets/25298812/21925069-d65c-43ec-a2d3-aabf24627d5e)
Scaled using cubic kernel

![example-600-l2](https://github.com/lovell/sharp/assets/25298812/67c9c74f-143c-4a70-b90d-a7465ced9845)
Scaled using lanczos2 kernel

**Comment by lovell** [comment]
Yes, the underlying logic that maps downsampling kernels to upsampling interpolators within libvips can be seen at:

https://github.com/libvips/libvips/blob/c2ede6c110ffcdb47606e8366b8f6a5ff7515b76/libvips/resample/resize.c#L113-L132

However this mapping was missing from the sharp documentation, which commit https://github.com/lovell/sharp/commit/6257994746652f1d337f1601f796ed020df3bbdf addresses (as well as exposing the missing `bilinear` kernel). Thanks for reporting.

**Comment by einarakerlind** [comment]
Thanks for clarifying!

**Comment by lovell** [comment]
v0.33.4 now available.
